amulet/sentry.py

```diff
@@ -254,24 +261,26 @@ def wait_for_status(self, juju_env, services, timeout=300):
             raise
 
     def wait(self, timeout=300):
-        #for unit in self.unit:
         ready = False
         try:
             with helpers.timeout(timeout):
                 # Make sure we're in a 'started' state across the board
-                waiter.wait(timeout=timeout)
-                while not ready:
+                now = time.time()
+                while not ready and now + timeout < time.time():
+                    waiter.wait(timeout=15)
                     for unit in self.unit.keys():
-                        status = self.unit[unit].juju_agent()
-                        # Check if we have a hook key and it's not None
-                        if status is None:
-                            ready = False
-                            break
-                        if 'hook' in status and status['hook']:
-                            ready = False
-                            break
-                        else:
-                            ready = True
+                        if unit['agent-state'] == 'started':
+                            status = self.unit[unit].juju_agent()
+
+                            # Check if we have a hook key and it's not None
+                            if status is None:
+                                ready = False
+                                break
+                            if 'hook' in status and status['hook']:
```

Comment:
If you mean the 'hook' in status, that was existing. I'm not exactly sure what the use case for it is, so I left it as is.